File: 125-ValidPalindrome/javascript/solution.js
// Last updated: 3/6/2026, 12:46:06 AM
/**
 * @param {string} s
 * @return {boolean}
 */
var isPalindrome = function(s) {
    s = s.toLowerCase().trim();
    console.log(s)
    let smooth = ''
    for (let i of s){
        if(i>='a' && i<='z'||i>='0' && i<='9'){
            smooth+=i
        }
    }
    return smooth.split('').reverse().join('') === smooth
};

File: 125-ValidPalindrome/cpp/solution.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        for (char& c : s) {
            c = tolower(c);
        }
        string smooth = "";
        for (char i : s) {
            if ((i >= 'a' && i <= 'z') || (i >= '0' && i <= '9')) {
                smooth += i;
            }
        }
        int l = 0, r = smooth.size() - 1;
        while (l <= r) {
            if (smooth[l] != smooth[r])
                return false;
            ++l;
            --r;
        }
        return true;
    }
};